src/ui/menu.c

```diff
@@ -111,6 +114,98 @@ UX_STEP_CB(ux_menu_back_step, pb, ui_menu_main(), {&C_icon_back, "Back"});
 // #2 screen: back button to main menu
 UX_FLOW(ux_menu_about_flow, &ux_menu_info_step, &ux_menu_back_step, FLOW_LOOP);
 
-void ui_menu_about() {
+static void ui_menu_about() {
     ux_flow_init(0, ux_menu_about_flow, NULL);
 }
+
+#else
+
+static const char* const info_types[] = {"Version", APPNAME};
+static const char* const info_contents[] = {APPVERSION, "(c) 2022 Ledger"};
+
+static void quit_app_callback(void) {
+    releaseContext();
+    os_sched_exit(-1);
+}
+
+#define NB_SETTINGS_SWITCHES 1
+static nbgl_layoutSwitch_t G_switches[NB_SETTINGS_SWITCHES];
+
+enum {
+    SWITCH_CONTRACT_DATA_SET_TOKEN = FIRST_USER_TOKEN,
+};
+
+static bool settings_nav_callback(uint8_t page, nbgl_pageContent_t* content) {
+    if (page == 0) {
+        content->type = SWITCHES_LIST;
+        content->switchesList.nbSwitches = NB_SETTINGS_SWITCHES;
+        content->switchesList.switches = (nbgl_layoutSwitch_t*) G_switches;
+    } else if (page == 1) {
+        content->type = INFOS_LIST;
+        content->infosList.nbInfos = ARRAY_COUNT(info_types);
+        content->infosList.infoTypes = (const char**) info_types;
+        content->infosList.infoContents = (const char**) info_contents;
+    } else {
+        return false;
+    }
+
+    return true;
+}
+
+static void ui_menu_main_nbgl(void);
+void ui_menu_settings(void);
+
+static void settings_controls_callback(int token, uint8_t index) {
+    bool new_setting;
+    UNUSED(index);
+    switch (token) {
+        case SWITCH_CONTRACT_DATA_SET_TOKEN:
+            G_switches[0].initState = !(G_switches[0].initState);
+            new_setting = (G_switches[0].initState == ON_STATE);
+            nvm_write((void*) &N_storage.scriptsAllowed, &new_setting, 1);
```

Comment:
In Tron there are handy little macros for toggling and getting the setting values. Up to you but it would be easier to use something similar if in the future there are other settings added.

```
#ifndef SETTINGS_H
#define SETTINGS_H

#include <stdint.h>

typedef uint8_t internal_storage_t;

#define N_settings (*(volatile internal_storage_t *)PIC(&N_storage_real))

// the settings, stored in NVRAM. Initializer is ignored by ledger.
extern const internal_storage_t N_storage_real;

// flip a bit k = 0 to 7 for u8
#define _FLIP_BIT(n, k)  (((n) ^ (1 << (k))))

// toggle a setting item
#define SETTING_TOGGLE(_set) do {\
    internal_storage_t _temp_settings = _FLIP_BIT(N_settings, _set); \
    nvm_write((void*)&N_settings, (void*)&_temp_settings, sizeof(internal_storage_t)); \
  } while(0)

// check a setting item
#define HAS_SETTING(k)  ((N_settings & (1 << (k))) >> (k))


#define S_DATA_ALLOWED     0
#define S_CUSTOM_CONTRACT  1
#define S_TRUNCATE_ADDRESS 2
#define S_SIGN_BY_HASH     3

#define S_INITIALIZED      7


#endif
```

Reply:
True but I'll let the devs of the application factorize their settings if they want so, I think it's a bit outside of the stax porting

Reply:
Ok